### ISSUE TITLE
Variables defined in Rust aren't documented properly

### ISSUE THREAD

**Issue by nickdrozd** [Wilfred/remacs#876]
Here's what I get running `(describe-variable 'last-command)`:

```
last-command’s value is ‘quit-window’
It is a terminal-local variable; global value is the same.

  This variable may be risky if used as a file-local variable.

Documentation:
Not documented as a variable.

[back]
```

`last-command` is defined in `keyboard.rs` along with a few other variables that suffer the same problem.

Here's what it looks like in GNU Emacs:

```
last-command is a variable defined in ‘keyboard.c’.
Its value is ‘eval-expression’
It is a terminal-local variable; global value is the same.

  This variable may be risky if used as a file-local variable.

Documentation:
The last command executed.
Normally a symbol with a function definition, but can be whatever was found
in the keymap, or whatever the variable ‘this-command’ was set to by that
command.

The value ‘mode-exit’ is special; it means that the previous command
read an event that told it to exit, and it did so and unread that event.
In other words, the present command is the event that made the previous
command exit.

The value ‘kill-region’ is special; it means that the previous command
was a kill command.

‘last-command’ has a separate binding for each terminal device.
See Info node ‘(elisp)Multiple Terminals’.
```

`Remacs 26.0.60 (build 2, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2018-09-25`

**Comment by shaleh** [comment]
https://github.com/Wilfred/remacs/blob/cdcf51bae37a10f4c3a1b78252c70ab0afc773f8/rust_src/remacs-lib/docfile.rs#L176

`last-command` is marked as a keyboard so it is ignored by the code linked above.

**Comment by nickdrozd** [comment]
`header-line-format` from `buffers.rs` also doesn't work.

**Comment by nickdrozd** [comment]
`(describe-variable 'post-self-insert-hook)` does work properly, but `(find-definition-noselect 'post-self-insert-hook 'defvar)` fails with a type error because `find-function-rust-source` doesn't know how to look for variables.

**Comment by nickdrozd** [comment]
Different signatures:

`(find-function-rust-source IDENTIFIER FILE)`
`(find-function-C-source FUN-OR-VAR FILE TYPE)`

**Comment by db48x** [comment]
Yes, I simply never implemented this for `kboard` or `per_buffer` variables. Unfortunately, I don't recall why I was skipping them; looking at it again I think that `docfile.c` is simply confusing. When I look at the generated `DOC` file in a normal Emacs build, I see that there's nothing special about these types of variables in the documentation.

**Comment by nickdrozd** [comment]
This should stay open until https://github.com/Wilfred/remacs/pull/880 is merged.
